envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/MetadataNodeGroup.kt

```diff
@@ -64,11 +69,38 @@ class MetadataNodeGroup(val properties: SnapshotProperties) : NodeGroup<Group> {
         return ListenersHostPortConfig(ingressHost, ingressPort, egressHost, egressPort)
     }
 
+    private fun metadataToAccessLogFilterHttpCode(metadata: Struct): AccessLogFilterHttpCode? {
+        val httpFilterCode = metadata.fieldsMap["access_log_filter_http_code"]?.stringValue
```

Comment:
I would change a structure of metadata:
```
access_log_filter:
  status_code_filter: 
    status_code:
    operator:
```
So in future we will have elastic structure which allows to add new filters(header, flags, ..)

Reply:
As discussed today, let's go with this:
```
access_log_filter:
  status_code_filter: "ge:400"
```

We get raw string, colon separated. We split by colon and extract `status_code` and `operator` from that.
In the future if required we can add `header_filter` or extend `status_code_filter` by supporting comma-separated values.

Reply:
done